src/site/content/en/blog/airshift-perf-optimization/index.md

```diff
@@ -0,0 +1,250 @@
+---
+title: "5 ways AirSHIFT made their React app responsive"
+subhead: |
+  A real world case study of React SPA performance optimization.
+date: 2019-10-25
+authors:
+  - uskay
+  - yosuke_furukawa
+  - satoshi_arai
+  - kento_tsuji
+hero: hero.png
+alt: A logo of AirSHIFT
+description: |
+  This case study explains how the AirSHIFT team optimized the responsiveness
+  of their desktop React SPA and satisfied their users.
+tags:
+  - post
+  - fast
+---
+
+Website performance is not just about load time. It is critical to provide a fast and responsive experience to users, especially for productivity desktop apps which people use everyday. The engineering team at [Recruit Technologies](https://recruit-tech.co.jp/) went through a refactoring project to improve one of their Web apps, [AirSHIFT](https://airregi.jp/shift/), for better user input performance. Here's how they did it.
+
+## Slow response, less productivity
+AirSHIFT is a desktop Web application that helps store owners, like restaurants and cafes, to manage the shift work of their staff members. Built with React, the Single Page Application provides rich client features including various grid tables of shift schedules cut by day, week, month and more.
+
+<img class="w-screenshot" src="airshift_visual.png" alt="A visual image of AirSHIFT">
+
+To meet user expectations they have been adding new functionalities to the app but as they continue to add features, they started seeing more feedback around slow performance. The engineering manager of the product, Yosuke Furukawa says "We did a user research and one of the store owners even said she would leave her seat to brew coffee after clicking a button… just to kill time waiting for the shift table to load, which was a shocking fact we faced". 
+
+After going through the research, the engineering team realized that many of their users were still using low spec computers such as 1 GHz Celeron M laptop from 10 years ago with a tiny amount of memory and having a massive shift table on screen. 
+
+<figure class="w-figure w-figure--fullbleed">
+  <video controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/web-dev-assets/airshift-perf-optimization/endless_spinner_vp9.webm" type="video/webm; codecs=vp8">
+    <source src="https://storage.googleapis.com/web-dev-assets/airshift-perf-optimization/endless_spinner_h264.mp4" type="video/mp4; codecs=h264">
+  </video>
+ <figcaption class="w-figcaption w-figcaption--fullbleed">
+    Endless spinner on low end devices
+  </figcaption>
+</figure>
+
+On the other hand, AifSHIFT had many expensive scripts that ran on the device, which was consuming the majority of the main thread activity, but the engineering team did not realize how harmful it was because they were developing and testing on a rich spec computer with fast WiFi.
```

Comment:
```suggestion
The AirSHIFT app was blocking the main thread with expensive scripts,
but the engineering team didn't realize how expensive the scripts were because they
were developing and testing on rich spec computers with fast Wi-Fi connections.
```